resources.go

```diff
@@ -1293,3 +1244,192 @@ func FromName(options AutoNameOptions) func(res *tfbridge.PulumiResource) (inter
 		return vs, nil
 	}
 }
+
+func renameLegacyModules(prov *tfbridge.ProviderInfo) {
+
+	renameResourceWithAlias := func(
+		tfName string,
+		tokName string,
+		newTokName string,
+		legacyModule string,
+		currentModule string,
+		info *tfbridge.ResourceInfo) {
+
+		legacyTFName := tfName + "_legacy"
+
+		if info == nil {
+			info = &tfbridge.ResourceInfo{}
+		}
+		legacyInfo := *info
+		currentInfo := *info
+
+		legacyInfo.Tok = azureResource(legacyModule, tokName)
+		legacyType := legacyInfo.Tok.String()
+
+		if newTokName != "" {
+			tokName = newTokName
+		}
+
+		currentInfo.Tok = azureResource(currentModule, tokName)
+		currentInfo.Aliases = []tfbridge.AliasInfo{
+			{Type: &legacyType},
+		}
+
+		if legacyInfo.Docs == nil {
+			legacyInfo.Docs = &tfbridge.DocInfo{
+				Source: tfName[len("azurerm_"):] + ".html.markdown",
+			}
+		}
+
+		prov.Resources[tfName] = &currentInfo
+		prov.Resources[legacyTFName] = &legacyInfo
+		prov.P.ResourcesMap[legacyTFName] = prov.P.ResourcesMap[tfName]
+	}
+
+	renameDataSourceWithAlias := func(
+		tfName string,
+		tokName string,
+		newTokName string,
+		legacyModule string,
+		currentModule string,
+		info *tfbridge.DataSourceInfo) {
+
+		legacyTFName := tfName + "_legacy"
+
+		if info == nil {
+			info = &tfbridge.DataSourceInfo{}
+		}
+		legacyInfo := *info
+		currentInfo := *info
+
+		legacyInfo.Tok = azureDataSource(legacyModule, tokName)
+
+		if newTokName != "" {
+			tokName = newTokName
+		}
+
+		currentInfo.Tok = azureDataSource(currentModule, tokName)
+
+		if legacyInfo.Docs == nil {
+			legacyInfo.Docs = &tfbridge.DocInfo{
+				Source: tfName[len("azurerm_"):] + ".html.markdown",
+			}
+		}
+
+		prov.DataSources[tfName] = &currentInfo
+		prov.DataSources[legacyTFName] = &legacyInfo
+		prov.P.DataSourcesMap[legacyTFName] = prov.P.DataSourcesMap[tfName]
+	}
+
+	// New Authorization Mod - this combines the old MSI and Role Modules
+	renameResourceWithAlias("azurerm_role_assignment", "Assignment", "", azureLegacyRole, azureAuthorization,
+		&tfbridge.ResourceInfo{
+			Fields: map[string]*tfbridge.SchemaInfo{
+				// Suppress auto-naming of this field. It is autonamed to a GUID in the underlying provider.
+				azureName: {Name: azureName},
+			},
+		},
+	)
+	renameResourceWithAlias("azurerm_role_definition", "Definition", "", azureLegacyRole, azureAuthorization, nil)
+	renameResourceWithAlias("azurerm_user_assigned_identity", "UserAssignedIdentity", "", azureLegacyMSI,
+		azureAuthorization, &tfbridge.ResourceInfo{
+			Docs: &tfbridge.DocInfo{
+				Source: "user_assigned_identity.markdown",
+			},
+		},
+	)
+	renameDataSourceWithAlias("azurerm_role_definition", "getRoleDefinition", "",
```

Comment:
this is def confusing since it's getRoleDefinition here, but Definition above.

Reply:
done